src/rgw/rgw_asio_frontend.cc

```diff
@@ -1155,14 +1125,16 @@ void AsioFrontend::unpause()
 class RGWAsioFrontend::Impl : public AsioFrontend {
  public:
   Impl(RGWProcessEnv& env, RGWFrontendConfig* conf,
-       rgw::dmclock::SchedulerCtx& sched_ctx)
-    : AsioFrontend(env, conf, sched_ctx) {}
+       rgw::dmclock::SchedulerCtx& sched_ctx,
+       ceph::async::io_context_pool& context_pool)
+    : AsioFrontend(env, conf, sched_ctx, context_pool) {}
 };
 
 RGWAsioFrontend::RGWAsioFrontend(RGWProcessEnv& env,
                                  RGWFrontendConfig* conf,
-				 rgw::dmclock::SchedulerCtx& sched_ctx)
-  : impl(new Impl(env, conf, sched_ctx))
+				 rgw::dmclock::SchedulerCtx& sched_ctx,
+				 ceph::async::io_context_pool& context_pool)
+  : impl(new Impl(env, conf, sched_ctx, context_pool))
```

Comment:
does the frontend still need the io_context_pool, now that AppMain handles the shutdown? it looks like this could just take a reference to io_context